hazelcast-client/src/main/java/com/hazelcast/client/config/ClientIcmpPingConfig.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+/**
+ * ICMP ping is used to detect if machine that a remote hazelcast member runs on alive or not
+ * ICMP Ping detector will attempt 3 `max-attempts` pings, one every 1000 millis `interval-milleseconds`
+ * and will wait up-to 1000 millis `timeout-millisecond` for each to complete.
+ * If after 3 `max-attempts` seconds, there was no successful ping, the member will get suspected.
+ */
+public class ClientIcmpPingConfig {
+
+    private static final int DEFAULT_TIMEOUT_MILLISECONDS = 1000;
+    private static final int DEFAULT_INTERVAL_MILLISECONDS = 1000;
+    private static final int DEFAULT_TTL = 255;
+    private static final int DEFAULT_MAX_ATTEMPT = 3;
+
+    private int timeoutMilliseconds = DEFAULT_TIMEOUT_MILLISECONDS;
+    private int intervalMilliseconds = DEFAULT_INTERVAL_MILLISECONDS;
+    private boolean echoFailFastOnStartup = true;
+    private int ttl = DEFAULT_TTL;
+    private int maxAttempts = DEFAULT_MAX_ATTEMPT;
+    private boolean enabled;
+
+    /**
+     * This cannot be more than the interval value. Should always be smaller.
+     *
+     * @return timeoutMilliseconds Ping timeout in milliseconds.
+     */
+    public int getTimeoutMilliseconds() {
+        return timeoutMilliseconds;
+    }
+
+    /**
+     * This cannot be more than the interval value. Should always be smaller.
+     *
+     * @param timeoutMilliseconds Ping timeout in milliseconds.
+     * @return this ClientIcmpPingConfig instance for chaining
+     */
+    public ClientIcmpPingConfig setTimeoutMilliseconds(int timeoutMilliseconds) {
+        this.timeoutMilliseconds = timeoutMilliseconds;
+        return this;
+    }
+
+    /**
+     * Default and min allowed, 1000 millis
```

Comment:
Aren't these javadocs a bit cryptic? It's not clear what _Default and min allowed, 1000 millis_ means and it doesn't explain what the method does.